src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isEmpty()) {
+            throw new AuthenticationException("Can't authenticate user");
+        }
+        User user = userFromDbOptional.get();
+        if (user.getPassword().equals(HashUtil.hashPassword(password, user.getSalt()))) {
+            return user;
+        } else {
+            throw new AuthenticationException("Wrong password!");
+        }
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
```

Comment:
It's better to create separate method
```suggestion
    public User register(String email, String password) throws RegistrationException {
        validateRegisterData(email, password);
        return userService.add(new User(email, password));
```

```
private void validateRegisterData(String email, String password) throws RegistrationException {
        if (email == null || email.isBlank()
                || password == null || password.isBlank()) {
            throw new RegistrationException("Email or password is blank");
        }
        if (userService.findByEmail(email).isPresent()) {
            throw new RegistrationException("User with this email already exists");
        }
    }
```